api/tests/routes/native/v1/bookings_test.py

```diff
@@ -160,6 +161,64 @@ def test_post_non_free_bookings_or_from_wrong_subcategories_without_archive(
         assert booking.status == BookingStatus.CONFIRMED
         assert not booking.dateUsed
 
+    @override_features(ENABLE_CHARLIE_BOOKINGS_API=True)
+    @freeze_time("2022-10-12 17:09:25")
+    @patch("pcapi.tasks.external_api_booking_notification_tasks.external_api_booking_notification_task.delay")
+    def test_bookings_send_notification_to_external_api_with_external_event_booking(self, mocked_task, client):
+        base_url = "https://book_my_offer.com"
+        external_notification_url = base_url + "/notify"
+        user = users_factories.BeneficiaryGrant18Factory(
+            email=self.identifier, dateOfBirth=datetime(2007, 1, 1), phoneNumber="+33101010101"
+        )
+        provider = providers_factories.ProviderFactory(
+            name="Technical provider",
+            localClass=None,
+            notificationExternalUrl=external_notification_url,
+        )
+        providers_factories.OffererProviderFactory(provider=provider)
+        stock = EventStockFactory(
+            lastProvider=provider,
+            priceCategory__price=2,
+            offer__subcategoryId=subcategories.SEANCE_ESSAI_PRATIQUE_ART.id,
+            offer__lastProvider=provider,
+            offer__venue__address="1 boulevard Poissonniere",
+            offer__extraData={"ean": "1234567890123"},
+            idAtProviders="",
+            dnBookedQuantity=14,
+            quantity=20,
+        )
+
+        response = client.with_token(self.identifier).post(
+            "/native/v1/bookings",
+            json={"stockId": stock.id, "quantity": 1},
+        )
+
+        assert response.status_code == 200
+
+        booking = Booking.query.filter(Booking.stockId == stock.id).first()
+        assert booking.userId == user.id
+        assert response.json["bookingId"] == booking.id
+        assert booking.status == BookingStatus.CONFIRMED
+        assert not booking.dateUsed
+
+        assert mocked_task.call_count == 1
+        data = mocked_task.call_args.args[0].__dict__["data"].__dict__
+        assert data["offer_ean"] == "1234567890123"
```

Comment:
`__dict__` est un attribut privé, n'y a-t-il pas un autre moyen d'accéder aux données ? Il me semble que le code suivant fonctionne (et serait plus lisible) : 

```suggestion
        notification = mocked_task.call_args.args[0].data
        assert data.offfer_ean == "1234567890123"
```

Tu noteras aussi qu'on peut accéder directement aux attributs de `ExternalApiBookingNotificationRequest` , ce qui évite la lourdeur d'un accès via `__dict__`.